### PR TITLE
Add timeout reasons to various modules

### PR DESCRIPTION
**Pull request by alazymeme**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable
- [ ] Documentation in docs/ or install-docs/ was updated, if applicable

Fixes https://github.com/pajbot/pajbot/issues/843

### CODE REVIEW

**Comment by RAnders00** [comment]
Why did you move settings in order?

**Comment by alazymeme** [comment]
For some of the modules the checkboxes were halfway down or at the bottom/top. I've moved the checkboxes (where necessary) up the top for neatness and consistency.

**Comment by RAnders00** [comment]
I would order settings by relevance, most commonly changed or viewed at the top.

**Comment by alazymeme** [comment]
I don't believe that's a good idea: what is relevant for one person, may not be relevant for another person. Nobody should be deciding that for other people either.

**Comment by alazymeme** [comment]
Before this gets merged, I just need a little bit of help with the link checker module and ascii modules. I noticed that there are two timeout reasons configured. Why is this? Am I able to parse through the parameters to a custom reason? (Eg: `{punishment}`)

File: pajbot/modules/emote_timeout.py [hunk 1]
```diff
@@ -56,6 +56,42 @@ class EmoteTimeoutModule(BaseModule):
...
+            default="No BTTV emotes allowed",
+            constraints={},
+        ),
+        ModuleSetting(
+            key="emoji_timeout_reason",
+            label="Emoji Timeout Reason",
+            type="text",
+            required=False,
+            placeholder="",
+            default="No emojis allowed",
```
Comment (RAnders00) [inline pajbot/modules/emote_timeout.py:92]:
```suggestion
            default="No emoji allowed",
```

**Comment by RAnders00** [comment]
For the replacements, you can try (e.g.):

https://github.com/pajbot/pajbot/blob/c18b5940c325492845be8a4ba77214a061775ca5/pajbot/modules/ascii.py#L92-L94

```python
            self.bot.whisper(
                source, self.settings["reason_whisper_msg"].format(punishment=punishment)
            )
```

and add a setting that has `You have been {punishment} because your message contained too many ascii characters.` as its default value.

**Comment by alazymeme** [comment]
> 
> 
> For the replacements, you can try (e.g.):
> 
> https://github.com/pajbot/pajbot/blob/c18b5940c325492845be8a4ba77214a061775ca5/pajbot/modules/ascii.py#L92-L94
> 
> ```python
>             self.bot.whisper(
>                 source, self.settings["reason_whisper_msg"].format(punishment=punishment)
>             )
> ```
> 
> and add a setting that has `You have been {punishment} because your message contained too many ascii characters.` as its default value.

Ok I'll apply this to other modules. The question still stands: For the links module, https://github.com/pajbot/pajbot/issues/843 seems to be applicable. Are we able to remove these lines? What function do they serve?

https://github.com/pajbot/pajbot/blob/127298207742e94904b357a7a12f3acc7aa0d90f/pajbot/modules/linkchecker.py#L252-L255

**Comment by RAnders00** [comment]
> > For the replacements, you can try (e.g.):
> > https://github.com/pajbot/pajbot/blob/c18b5940c325492845be8a4ba77214a061775ca5/pajbot/modules/ascii.py#L92-L94
> > 
> > ```python
> >             self.bot.whisper(
> >                 source, self.settings["reason_whisper_msg"].format(punishment=punishment)
> >             )
> > ```
> > 
> > 
> > and add a setting that has `You have been {punishment} because your message contained too many ascii characters.` as its default value.
> 
> Ok I'll apply this to other modules. The question still stands: For the links module, #843 seems to be applicable. Are we able to remove these lines? What function do they serve?
> 
> https://github.com/pajbot/pajbot/blob/127298207742e94904b357a7a12f3acc7aa0d90f/pajbot/modules/linkchecker.py#L252-L255

I'm not sure how the link checker works internally, I think it's a lot of spaghetti in there. I can't say for sure without taking a closer look

**Comment by alazymeme** [comment]
@pajlada Looks good to me - works as functioned and fixes the issue as well

**Pull request merged**